libs/auth-api-lib/src/lib/services/delegations.service.ts

```diff
@@ -352,23 +363,126 @@ export class DelegationsService {
+
+    const [delegations, needsCheckDelegations] = await Promise.all([
+      Promise.all(delegationPromises),
+      Promise.all(needVerifyDelegationPromises),
+    ])
+
+    // Only check live status, i.e. dead or alive for needsCheckDelegations
+    const {
+      aliveDelegations,
+      deceasedDelegations,
+    } = await this.getLiveStatusFromDelegations(needsCheckDelegations.flat())
+
+    if (deceasedDelegations.length > 0) {
+      // Invalidate all deceased delegations
+      await this.invalidateDelegations(user, deceasedDelegations)
+
+      this.auditService.auditSystem({
+        action: 'deceasedDelegation',
+        resources: deceasedDelegations.map(({ id }) => id).filter(isDefined),
+      })
     }
-    const delegationSets = await Promise.all(delegationPromises)
 
     return uniqBy(
-      ([] as DelegationDTO[]).concat(...delegationSets),
+      [...delegations.flat(), ...aliveDelegations],
       'fromNationalId',
     ).filter((delegation) => delegation.fromNationalId !== user.nationalId)
   }
 
+  /**
+   * Invalidates delegations by updating delegation.scope[index].validTo date field to start of yesterday.
+   */
+  private async invalidateDelegations(
+    user: User,
+    deceasedDelegations: DelegationDTO[],
+  ) {
+    const delegations = await Promise.all(
+      // We need to refetch the delegation to make sure we get all delegation.scopes
+      deceasedDelegations.map(({ id }) =>
+        id ? this.findById(user, id, true) : undefined,
+      ),
+    )
+    const delegationsWithAllScopes = delegations.filter(isDefined)
+
+    if (delegationsWithAllScopes.length > 0) {
+      // Gather all delegation scope ids that needs to be invalidated
+      const delegationScopeIds = delegationsWithAllScopes
+        .map((delegation) =>
+          delegation.scopes?.map(({ id }) => id).filter(isDefined),
+        )
+        .filter(isDefined)
+        .flat()
+
+      if (delegationScopeIds.length > 0) {
+        await Promise.all(
+          delegationScopeIds.map((id) =>
+            this.delegationScopeService.invalidate(id),
+          ),
+        )
+      }
+    }
+  }
+
+  /**
+   * Divides delegations into alive and deceased delegations
+   * - Makes calls for every delegation to NationalRegistry to check if the person exists.
+   * - Divides the delegations into alive and deceased delegations, based on
+   *   1. All companies will be divided into alive delegations.
+   *   2. If the person exists in NationalRegistry, then the delegation is alive.
+   */
+  private async getLiveStatusFromDelegations(
+    delegations: DelegationDTO[],
+  ): Promise<{
+    aliveDelegations: DelegationDTO[]
+    deceasedDelegations: DelegationDTO[]
+  }> {
+    if (delegations.length === 0) {
+      return {
+        aliveDelegations: [],
+        deceasedDelegations: [],
+      }
+    }
+
+    const delegationsPromises = delegations.map(({ fromNationalId }) =>
+      this.nationalRegistryClient.getIndividual(fromNationalId),
+    )
```

Comment:
A few suggestions:

1. Should not try and look up company ids in national registry.
2. If an error is thrown during lookup, we may want to swallow it and assume the delegation is alive.

Reply:
Great call out, here is the updated version
```
 /**
   * Divides delegations into alive and deceased delegations
   * - Makes calls for every delegation to NationalRegistry to check if the person exists.
   * - Divides the delegations into alive and deceased delegations, based on
   *   1. All companies will be divided into alive delegations.
   *   2. If the person exists in NationalRegistry, then the delegation is alive.
   */
  private async getLiveStatusFromDelegations(
    delegations: DelegationDTO[],
  ): Promise<{
    aliveDelegations: DelegationDTO[]
    deceasedDelegations: DelegationDTO[]
  }> {
    if (delegations.length === 0) {
      return {
        aliveDelegations: [],
        deceasedDelegations: [],
      }
    }

    const delegationsPromises = delegations
      // Filter out companies, since we do not need to make a natinonal registry call for them.
      .filter(({ fromNationalId }) => !kennitala.isCompany(fromNationalId))
      .map(({ fromNationalId }) =>
        this.nationalRegistryClient.getIndividual(fromNationalId),
      )

    try {
      // Check if delegations is linked to a person, i.e. not deceased
      const persons = await Promise.all(delegationsPromises)
      const personsValues = persons.filter(isDefined)

      // Divide delegations into alive or deceased delegations.
      const [aliveDelegations, deceasedDelegations] = partition(
        delegations,
        ({ fromNationalId }) =>
          // All companies will be divided into aliveDelegations
          kennitala.isCompany(fromNationalId) ||
          // Make sure we can match the person to the delegation, i.e. not deceased
          personsValues.find((person) => person?.nationalId === fromNationalId),
      )

      return {
        aliveDelegations,
        deceasedDelegations,
      }
    } catch (error) {
      this.logger.error(
        `Error getting live status from delegations. Delegations: ${delegations.map(
          (d) => d.id,
        )}`,
        error,
      )

      // We do not want to fail the whole request if we cannot get the live status from delegations.
      // Therefore we return all delegations as alive delegations.
      return {
        aliveDelegations: delegations,
        deceasedDelegations: [],
      }
    }
  }
```

2.